dynamic-tools/zap-extensions/zapwso2jiraplugin/ExtensionJiraIssueCreater.java

```diff
@@ -0,0 +1,199 @@
+     * @param issueLabel   This is a custom feild defined in wso2 jira
+     * @param filePath-    This is the file path where ZAP generates its report. This includes file name too.
+     * @return
+     */
+
+    public void createJiraIssues(String jiraBaseURL, String jiraUserName, String jiraPassword, String projectKey,
+            String asssignee, String product, String issueLabel, String filePath, String workspace,
+            String backupFolder) {
+
+        String issue;
+        String credentials = jiraUserName + ":" + jiraPassword;
+        String auth = new String(Base64.encodeBase64(credentials.getBytes()));
+        String BASE_URL = jiraBaseURL;
+
+        log.info("Backup Folder Path" + backupFolder);
+
+        String version = MavenBuildReceiver.getPomVersion(new File(workspace + "/pom.xml"));
+
+        String summary = "[JENKINS][" + product + "-" + version + "][DynamicScan-Nightly-Build-Report]";
+
+        //String summary = "[JENKINS][" + product + "][DynamicScan-Nightly-Build-Report]";
+
+        XmlDomParser xmlParser = new XmlDomParser();
+
+        //Checking if there is any issues exists in the ZAP report generated after the scan
+        boolean issueExist = xmlParser.isIssueExistsInReport();
+
+        //Checking if there is any already available issues reported in the jira
+        String issueKey = xmlParser.checkForIssueExistence(auth, BASE_URL, summary, projectKey);
+
+        String fileNewPath = xmlParser.reNameFile(product, filePath);
+
+        if (issueExist) {
+            try {
+                if (issueKey.equals("")) {
+                    String issueToBeCreated = xmlParser
+                            .createNewTicket(projectKey, asssignee, issueLabel, summary, product);
+                    issue = jira.invokePostMethod(auth, BASE_URL + "/rest/api/2/issue", issueToBeCreated);
+
+                    JSONObject createdIssue = new JSONObject(issue);
+                    issueKey = createdIssue.getString("key");
+                    log.info("Issue is created under : " + issueKey);
+
+                    jira.invokePostComment(auth, BASE_URL + "/rest/api/2/issue/" + issueKey + "/comment",
+                            xmlParser.createComment());
+
+                    jira.invokePutMethodWithFile(auth, BASE_URL + "/rest/api/2/issue/" + issueKey + "/attachments",
+                            xmlParser.compressFile(fileNewPath));
+                    log.info("Issue " + issueKey + " is updated with an attachment");
+                } else {
+
+                    updateJiraAttachments.modifyJiraContents(auth, BASE_URL, issueKey);
+
+                    jira.invokePostComment(auth, BASE_URL + "/rest/api/2/issue/" + issueKey + "/comment",
+                            xmlParser.createComment());
+                    log.info("Issue " + issueKey + " is updated with an attachment");
+
+                    jira.invokePutMethodWithFile(auth, BASE_URL + "/rest/api/2/issue/" + issueKey + "/attachments",
```

Comment:
"invokePostComment" and "invokePutMethodWithFile" are duplicated in 'if' and 'else'. We should be able to re-factor and avoid duplication.